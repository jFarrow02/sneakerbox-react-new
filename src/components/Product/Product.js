import React from 'react';
import './Product.scss';
import {
    Image,
} from '../../components';
import propTypes from 'prop-types';

const Product = props=>{
    const {
        slug,
        sku,
        modelName, 
        desc, 
        totalReviews,
        avgReview,
        pricing,
        tags,
        details, 
        onSale,
        sizes,
        isThumbnail,
    } = props;

    
    const now = Date.now();
    //const saleEndDate = pricing.to
    const baseUrl = '../../images/tn_images';
    const imgSrc = props.imageFiles.thumbnail;
    return(
        <div className='Product'>
            <h5>{modelName}</h5>
            <Image
                alt={desc}
                isThumbnail={isThumbnail}
                dynamicImgSrc={imgSrc}
                title={modelName}
            />
        </div>
    )
}

export default Product;

Product.propTypes = {
    desc: propTypes.string,
    imageFiles: propTypes.object,
    isThumbnail: propTypes.bool,
    modelName: propTypes.string,

};

Product.defaultProps = {
    desc: '',
    imageFiles: {
        thumbnail: '',
    },
    isThumbnail: false,
    modelName: '',
}

/***SCHEMA***/
    // _id: ObjectId,
    // key: String, -->'PRODUCT'
    // slug: String,
    // sku: String,
    // modelName: String,
    // desc: String,
    // totalReviews: Number,
    // avgReview: Number,
    // pricing: Object,
    // priceHistory: Object,
    // primaryCategory: ObjectId,
    // categories:[ObjectId],
    // tags: [String],
    // details: Object,
    //imageFiles: [{String}],
    // sizes: [{Number}],