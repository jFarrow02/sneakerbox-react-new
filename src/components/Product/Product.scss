@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';

@font-face{
    font-family: 'OpenSans';
    src: url($openSansReg);
}
.Product{
    
    // background-color: $offWhite;
    margin-left: 20px;
    border-right: 1px solid $smokyTopaz;
    border-left: 1px solid $smokyTopaz;
    padding: 20px;
    h4{
        font-family: 'OpenSans';
        margin-block-start: 0;
        margin-block-end: 0;
        margin-bottom: 10px;
        color: $darkRed;
    }
}