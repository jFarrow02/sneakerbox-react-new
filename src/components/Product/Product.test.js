import React from 'react';
import Product from './Product';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<Product/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});