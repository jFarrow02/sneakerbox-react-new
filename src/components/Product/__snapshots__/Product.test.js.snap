// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div
  className="Product"
>
  <h5>
    
  </h5>
  <img
    alt=""
    height=""
    src=""
    title=""
    width=""
  />
</div>
`;
